doc/tutorial/single_cell_detailed_recipe.rst

```diff
@@ -0,0 +1,419 @@
+Given the gid, this method returns the cell description which is the cell object passed to the constructor
+of the recipe. We return ``self.the_cell``.
+
+Step **(8)** overrides the :meth:`arbor.recipe.get_probes` method. It takes one argument: ``gid``.
+Given the gid, this method returns all the probes on the cell. The probes can be of many different kinds
+measuring different quantities on different locations of the cell. We pass these probes explicitly to the recipe
+and they are stored in ``self.the_probes``, so we return that variable.
+
+Step **(9)** overrides the :meth:`arbor.recipe.connections_on` method. It takes one argument: ``gid``.
+Given the gid, this method returns all the connections ending on that cell. These are typically synapse
+connections from other cell *sources* to specific *targets* on the cell with id ``gid``. Since we are
+simulating a single cell, and self-connections are not possible, we return an empty list.
+
+Step **(10)** overrides the :meth:`arbor.recipe.gap_junctions_on` method. It takes one argument: ``gid``.
+Given the gid, this method returns all the gap junctions on that cell. Gap junctions require 2 separate cells.
+Since we are simulating a single cell, we return an empty list.
+
+Step **(11)** overrides the :meth:`arbor.recipe.event_generators` method. It takes one argument: ``gid``.
+Given the gid, this method returns *event generators* on that cell. These generators trigger events (or
+spikes) on specific *targets* on the cell. They can be used to simulate spikes from other cells, to kick-start
+a simulation for example. Our cell uses a current clamp as a stimulus, and has no targets, so we return
+an empty list.
+
+Step **(12)** overrides the :meth:`arbor.recipe.global_properties` method. It takes one argument: ``kind``.
+This method returns the default global properties of the model which apply to all cells in the network of
+that kind. We return ``self.the_props`` which we defined in step **(1)**.
+
+.. Note::
+
+   You may wonder why the methods:  :meth:`arbor.recipe.num_sources`, :meth:`arbor.recipe.num_targets`,
+   and :meth:`arbor.recipe.cell_kind` are required, since they can be inferred by examining the cell description.
+   The recipe was designed to allow building simulations efficiently in a distributed system with minimum
+   communication. Some parts of the model initialization require only the cell kind, or the number of
+   sources and targets, not the full cell description which can be quite expensive to build. Providing these
+   descriptions separately saves time and resources for the user.
+
+   More information on the recipe can be found :ref:`here <modelrecipe>`.
+
+Now we can intantiate a ``single_recipe`` object using the ``cell`` and ``probe`` we created in the
+previous section:
+
+.. code-block:: python
+
+   # Instantiate recipe
+   # Pass the probe in a list because that it what single_recipe expects.
+   recipe = single_recipe(cell, [probe])
+
+The execution context
```

Comment:
Expansion on this section and the next could be in new tutorials in one of those 'lines'.

Reply:
I don't understand.

Reply:
I don't know how to refer to a particular review comment here, it's the first on `doc/tutorial/single_cell_model.rst`. There I mentioned the idea of the tutorial lines. I want to know a) what you think of that idea and b) if you agree that expanding on execution context and domain decomposition would be a good idea for one of those lines. This is of course not related at all to this review so can be taken elsewhere :)

Reply:
Oh Okay. I'm not sure I understand exactly what you mean, but I think expanding on the execution context and the domain decomposition is a good idea, maybe it can be done in the concept pages? It depends on what you have in mind, maybe the tutorial is a better spot.

Reply:
I assume the confusion is about what I mean with 'tutorial lines'. With a 'line', I mean a few successive tutorials building up in a particular direction, as in a line of reasoning. You started a 'line' here in the direction of more complex cells, another line could be about more complex networks and yet another line about more complex execution. This latter one one would probably include the expansion on execution context, domain decomposition. Is this clearer?

Yet more tutorials can then bring the lines together to get tutorials involving complexities in multiple 'directions'. Ultimately a large network of detailed cells running on HPC for instance.

I'll make a note in the docs wishlist. By the way, I think you can tick off a few things there?

Reply:
I assume the confusion is about what I mean with 'tutorial lines'. With a 'line', I mean a few successive tutorials building up in a particular direction, as in a line of reasoning. You started a 'line' here in the direction of more complex cells, another line could be about more complex networks and yet another line about more complex execution. This latter one one would probably include the expansion on execution context, domain decomposition. Is this clearer?

Yet more tutorials can then bring the lines together to get tutorials involving complexities in multiple 'directions'. Ultimately a large network of detailed cells running on HPC for instance.

I'll make a note in the docs wishlist. By the way, I think you can tick off a few things there?

Reply:
Okay, yes I understand better now. I think that's a good idea. 
I'll tick off things from that issue.